build/apple/download_swift_format.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env vpython3
+
+# Copyright (c) 2024 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import argparse
+import os
+import deps
+
+from brave_chromium_utils import wspath
+from deps_config import DEPS_PACKAGES_URL
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Download swift-format')
+    parser.add_argument('version')
+    args = parser.parse_args()
+
+    url = f'{DEPS_PACKAGES_URL}/swift-format/swift-format-{args.version}.zip'
+    # Overwrite Chromium's swift-format version as it very out of date and
+    # we will not be editing Chromium swift files to format anyways.
+    output_dir = wspath('//third_party/swift-format')
+    deps.DownloadAndUnpack(url, output_dir)
```

Comment:
this will download and unpack the archive on each `npm run sync`. You need to guard this with sha256 check to not redownload when our binary is already there. Maybe pass hash as another script parameter.

You need to ensure this works when you call `npm run sync -- --sync_chromium`. This will re-sync Chromium which will enforce Chromium to check `swift-format` and rewrite it into the original version which your dep then should rewrite to our version back (sha256 check should fail and you should call DownloadAndUnpack).

Reply:
Updated to include hash check to avoid redownloading every time.